index.js

```diff
@@ -493,6 +493,7 @@ function mergeAllOfSchema (location, schema, mergedSchema) {
   mergedSchema.$id = `merged_${randomUUID()}`
   ajvInstance.addSchema(mergedSchema)
   location.schemaId = mergedSchema.$id
+  location.jsonPointer = '#'
```

Comment:
Will it be problem of deeply nested schema?

Reply:
It should work with nested schemas. Can you provide an example of what you mean?

Reply:
I get my answer when I create the example.
Here comes one problem related to previous fix.

Recursive `$ref` will be a problem when using `allOf`.

```js
{
  $id: 'base',
  anyOf: [
    {
      type: 'object',
      allOf: [
        {
          properties: {
            a: {
              anyOf: [
                { const: 'A1' },
                { const: 'A2' }
              ]
            }
          }
        },
        {
          properties: {
            b: { const: 'B' }
          }
        },
        {
          properties: {
            c: {
              oneOf: [
                { $ref: 'base' },
                { type: 'null' }
              ]
            }
          }
        }
      ]
    }
  ]
}
```

Reply:
I guess we can fix this. But there are some other problems with merging schemas that we should also. We can't use deepMerge for merging. I was trying to make it better #453, but it's not enough. We can check how [json-schema-merge-allof](https://www.npmjs.com/package/json-schema-merge-allof) works.

In general, we have some places where we have troubles (if/then, oneOf, anyOf, allOf, type array), bc json schema format designed for validation and described input types and we use json schema for serialization and ask people to describe output types.

Reply:
I think it worth an issue for discussion.